functions/_common/grapherRenderer.ts

```diff
@@ -158,18 +158,17 @@ async function fetchFromR2(
     return primaryResponse
 }
 
-async function fetchAndRenderGrapherToSvg(
+interface FetchGrapherConfigResult {
+    grapherConfig: GrapherInterface | null
+    status: number
+    etag: string | undefined
+}
+
+export async function fetchGrapherConfig(
     slug: string,
-    options: ImageOptions,
-    searchParams: URLSearchParams,
     env: Env,
-    etag?: string
-) {
-    const grapherLogger = new TimeLogger("grapher")
-
-    const url = new URL(`/grapher/${slug}`, env.url)
-    const slugOnly = url.pathname.split("/").pop()
-
+    etag: string
```

Comment:
Should definitely be 
```suggestion
    etag?: string
```